sympy/matrices/matrices.py

```diff
@@ -3433,6 +3433,95 @@ def _eval_matrix_exp_jblock(self):
         from .sparsetools import banded
         return self.__class__(banded(size, bands))
 
+
+
+    def analytic_func(self, f, x, parameter = None):
+        """
+        Computes f(A) where A is a Square Matrix
+        and F is an univariate analytic function.
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol, Matrix, exp, S, log
+
+        >>> x = Symbol('x')
+        >>> m = Matrix([[S(5)/4, S(3)/4], [S(3)/4, S(5)/4]])
+        >>> f = log(x)
+        >>> m.analytic_func(f, x)
+        Matrix([
+        [     0, log(2)],
+        [log(2),      0]])
+
+        Parameters
+        ==========
+
+        f : Univariate Analytic Function
+        x : parameter of f
+        parameter : If Symbol('t') then it returns f(A*t) where t is parameter
+
+        """
+        from sympy import eye, zeros, diff
+
+        t = 1
+        if not parameter == None:
+            try:
```

Comment:
I would start with `parameter = sympify(parameter)`. Then `parameter` could also be a string: `parameter='t'`. There is no need for `try...except` block as the resulting parameter would have attribute `is_Symbol`.

Reply:
I'd not use `sympify` but use `_sympify`, and make user to define the symbol outside, rather than parsing the string inside.

Reply:
Creating the symbol here could simplify the use but I have no strong feelings either way.

Reply:
How to import _sympify